exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +74,2808 @@ func TestStart(t *testing.T) {
+	}
+	_, err = g.GetSpecifiedContractPosition(context.Background(), "BTC_USDT-20220826-32000-C")
+	if err != nil && !strings.Contains(err.Error(), "USER_NOT_FOUND") {
+		t.Errorf("%s GetSpecifiedContractPosition() error expecting %v, but found %v", g.Name, errInvalidOrMissingContractParam, err)
+	}
+}
+
+func TestGetUsersLiquidationHistoryForSpecifiedUnderlying(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetUsersLiquidationHistoryForSpecifiedUnderlying(context.Background(), "BTC_USDT", ""); err != nil {
+		t.Errorf("%s GetUsersLiquidationHistoryForSpecifiedUnderlying() error %v", g.Name, err)
+	}
+}
+
+func TestPlaceOptionOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	_, err := g.PlaceOptionOrder(context.Background(), OptionOrderParam{
+		Contract:    "BTC_USDT-20220902-18000-P",
+		OrderSize:   -1,
+		Iceberg:     0,
+		Text:        "-",
+		TimeInForce: "gtc",
+		Price:       100,
+	})
+	if err != nil && !strings.Contains(err.Error(), "INVALID_KEY") {
+		t.Errorf("%s PlaceOptionOrder() error %v", g.Name, err)
+	}
+}
+
+func TestGetOptionFuturesOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetOptionFuturesOrders(context.Background(), "", "", "", 0, 0, time.Time{}, time.Time{}); err != nil && !strings.Contains(err.Error(), "INVALID_KEY") {
+		t.Errorf("%s GetOptionFuturesOrders() error %v", g.Name, err)
+	}
+}
+
+func TestCancelOptionOpenOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	pairs, err := g.FetchTradablePairs(context.Background(), asset.Options)
+	if err != nil {
+		t.Skip(err)
+	}
+	if len(pairs) == 0 {
+		t.Skip("No tradable pairs found")
+	}
+	if _, err := g.CancelMultipleOptionOpenOrders(context.Background(), pairs[0], "", ""); err != nil && !strings.Contains(err.Error(), "INVALID_KEY") {
+		t.Errorf("%s CancelOptionOpenOrders() error %v", g.Name, err)
+	}
+}
+func TestGetSingleOptionOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetSingleOptionOrder(context.Background(), ""); err != nil && !errors.Is(errInvalidOrderID, err) {
+		t.Errorf("%s GetSingleOptionorder() expecting %v, but found %v", g.Name, errInvalidOrderID, err)
+	}
+	if _, err := g.GetSingleOptionOrder(context.Background(), "1234"); err != nil && !strings.Contains(err.Error(), "INVALID_KEY") {
+		t.Errorf("%s GetSingleOptionOrder() error %v", g.Name, err)
+	}
+}
+
+func TestCancelSingleOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.CancelOptionSingleOrder(context.Background(), "1234"); err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") && !strings.Contains(err.Error(), "INVALID_KEY") {
```

Comment:
Error:
```
GateIO CancelSingleOrder() error Delete "https://fx-api-testnet.gateio.wsoptions/orders/1234": dial tcp: lookup fx-api-testnet.gateio.wsoptions: no such host
```

Reply:
Seems like all options are failing for authenticated endpoints please go over them and test implementation.